// @marcapo/modulesession/src/interfaces/companydetailview.ts
export interface ICompanyDetailView {
	company: ICompany;
	employees: IEmployee[];
	primaryEmployee: IEmployee;
	primaryLocation: ILocation;
	locations: ILocation[];
}

export interface IOpeningHours {
	datesByAppointment: boolean;
	periods: IPeriod[];
	specialHourPeriods: ISpecialHourPeriod[];
	openDays: number[];
	enableSpecialHourPeriods: boolean;
	modificationTimestamp: number;
	commentRaw: string;
}

export interface IEndDate {
	day: number;
	month: number;
	year: number;
}

export interface IStartDate {
	day: number;
	month: number;
	year: number;
}
export interface ISpecialPeriod {
	close: string;
	open: string;
}
export interface ISpecialHourPeriod {
	comment: string;
	endDate: IEndDate;
	isClosed: boolean;
	periods: ISpecialPeriod[];
	startDate: IStartDate;
}

export interface IPeriod {
	day: number;
	open: IDayTime;
	close: IDayTime;
}

export interface IDayTime {
	day: number;
	time: string;
}

export interface ILocation {
	additionalFields: any;
	address: IDeliveryAddress;
	createdAt: number;
	createdBy: string;
	dataApproval: number;
	id: string;
	multipleOwner: boolean;
	name: string;
	office: boolean;
	officeType: string;
	oldBranchPersonKey: string;
	oldPersonKey: string;
	origin: string;
	shopUrl: string;
	updatedAt: number;
	updatedBy: string;
	updatedByIp: string;
	url: string;
	locationOpeningHours?: IOpeningHours;
}

export interface IEmployee {
	additionalFields: any;
	address: IDeliveryAddress;
	companyId: string;
	createdAt: number;
	createdBy: string;
	dataApproval: number;
	email: string;
	fax: string;
	firstName: string;
	id: string;
	lastName: string;
	locationCollection: string[];
	mobile: string;
	oldBranchPersonKey: string;
	oldPersonKey: string;
	origin: string;
	phone: string;
	position: string;
	primaryLocation: string;
	role: string;
	salutatory: string;
	shopUrl: string;
	title: string;
	updatedAt: number;
	updatedBy: string;
	updatedByIp: string;
	url: string;
	employeeOpeningHours?: IOpeningHours;
}

export interface ICompany {
	additionalFields: any;
	createdAt: number;
	createdBy: string;
	dataApproval: number;
	defaultDeliveryAddress: IDeliveryAddress;
	defaultInvoiceAddress: IDeliveryAddress;
	deliveryAddressCollection: IDeliveryAddress[];
	email: string;
	externalKey: string;
	groupId: string;
	groupPath: string[];
	groups: IGroup[];
	id: string;
	inactive: boolean;
	invoiceAddressCollection: IDeliveryAddress[];
	name: string;
	oldPersonKey: string;
	origin: string;
	primaryEmployee: string;
	primaryLocation: string;
	salesRegion: ISalesRegion;
	salesRegionId: string;
	salesTaxCity: string;
	salesTaxCompanyname: string;
	salesTaxId: string;
	secondName: string;
	shopUrl: string;
	updatedAt: number;
	updatedBy: string;
	updatedByIp: string;
	url: string;
	companyOpeningHours?: IOpeningHours;
}

export interface ISalesRegion {
	id: string;
	name: string;
	oldSalesRegionKey: string;
	primarySalesForceWorkerId: string;
	salesForceWorkerCollection: string[];
	shopUrl: string;
}

export interface IGroup {
	description: string;
	groupKey: string;
	id: string;
	locale: string;
	name: string;
	oldGroupKey: string;
	parentGroupId: string;
	shopUrl: string;
}

export interface IDeliveryAddress {
	city: string;
	coordinates: number[];
	country: string;
	createdAt: number;
	createdBy: string;
	id: string;
	street: string;
	updatedAt: number;
	updatedBy: string;
	updatedByIp: string;
	zip: string;
}
