// @marcapo/modulesession/lib/src/interfaces/countrycodes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountryCodes = void 0;
var CountryCodes;
(function (CountryCodes) {
    CountryCodes["AFGHANISTAN"] = "AF";
    CountryCodes["ALBANIA"] = "AL";
    CountryCodes["ALGERIA"] = "DZ";
    CountryCodes["AMERICAN SAMOA"] = "AS";
    CountryCodes["ANDORRA"] = "AD";
    CountryCodes["ANGOLA"] = "AO";
    CountryCodes["ANTARCTICA"] = "AQ";
    CountryCodes["ANTIGUA AND BARBUDA"] = "AG";
    CountryCodes["ARGENTINA"] = "AR";
    CountryCodes["ARMENIA"] = "AM";
    CountryCodes["ARUBA"] = "AW";
    CountryCodes["AUSTRALIA"] = "AU";
    CountryCodes["AUSTRIA"] = "AT";
    CountryCodes["AZERBAIJAN"] = "AZ";
    CountryCodes["BAHAMAS"] = "BS";
    CountryCodes["BAHRAIN"] = "BH";
    CountryCodes["BANGLADESH"] = "BD";
    CountryCodes["BARBADOS"] = "BB";
    CountryCodes["BELARUS"] = "BY";
    CountryCodes["BELGIUM"] = "BE";
    CountryCodes["BELIZE"] = "BZ";
    CountryCodes["BENIN"] = "BJ";
    CountryCodes["BERMUDA"] = "BM";
    CountryCodes["BHUTAN"] = "BT";
    CountryCodes["BOLIVIA"] = "BO";
    CountryCodes["BOSNIA AND HERZEGOVINA"] = "BA";
    CountryCodes["BOTSWANA"] = "BW";
    CountryCodes["BOUVET ISLAND"] = "BV";
    CountryCodes["BRAZIL"] = "BR";
    CountryCodes["BRITISH INDIAN OCEAN TERRITORY"] = "IO";
    CountryCodes["BRUNEI DARUSSALAM"] = "BN";
    CountryCodes["BULGARIA"] = "BG";
    CountryCodes["BURKINA FASO"] = "BF";
    CountryCodes["BURUNDI"] = "BI";
    CountryCodes["CAMBODIA"] = "KH";
    CountryCodes["CAMEROON"] = "CM";
    CountryCodes["CANADA"] = "CA";
    CountryCodes["CAPE VERDE"] = "CV";
    CountryCodes["CAYMAN ISLANDS"] = "KY";
    CountryCodes["CENTRAL AFRICAN REPUBLIC"] = "CF";
    CountryCodes["CHAD"] = "TD";
    CountryCodes["CHILE"] = "CL";
    CountryCodes["CHINA"] = "CN";
    CountryCodes["CHRISTMAS ISLAND"] = "CX";
    CountryCodes["COCOS (KEELING) ISLANDS"] = "CC";
    CountryCodes["COLOMBIA"] = "CO";
    CountryCodes["COMOROS"] = "KM";
    CountryCodes["CONGO"] = "CG";
    CountryCodes["CONGO, THE DEMOCRATIC REPUBLIC OF THE"] = "CD";
    CountryCodes["COOK ISLANDS"] = "CK";
    CountryCodes["COSTA RICA"] = "CR";
    CountryCodes["C\u00D4TE D'IVOIRE"] = "CI";
    CountryCodes["CROATIA"] = "HR";
    CountryCodes["CUBA"] = "CU";
    CountryCodes["CYPRUS"] = "CY";
    CountryCodes["CZECH REPUBLIC"] = "CZ";
    CountryCodes["DENMARK"] = "DK";
    CountryCodes["DJIBOUTI"] = "DJ";
    CountryCodes["DOMINICA"] = "DM";
    CountryCodes["DOMINICAN REPUBLIC"] = "DO";
    CountryCodes["ECUADOR"] = "EC";
    CountryCodes["EGYPT"] = "EG";
    CountryCodes["EL SALVADOR"] = "SV";
    CountryCodes["EQUATORIAL GUINEA"] = "GQ";
    CountryCodes["ERITREA"] = "ER";
    CountryCodes["ESTONIA"] = "EE";
    CountryCodes["ETHIOPIA"] = "ET";
    CountryCodes["FALKLAND ISLANDS (MALVINAS)"] = "FK";
    CountryCodes["FAROE ISLANDS"] = "FO";
    CountryCodes["FIJI"] = "FJ";
    CountryCodes["FINLAND"] = "FI";
    CountryCodes["FRANCE"] = "FR";
    CountryCodes["FRENCH GUIANA"] = "GF";
    CountryCodes["FRENCH POLYNESIA"] = "PF";
    CountryCodes["FRENCH SOUTHERN TERRITORIES"] = "TF";
    CountryCodes["GABON"] = "GA";
    CountryCodes["GAMBIA"] = "GM";
    CountryCodes["GEORGIA"] = "GE";
    CountryCodes["GERMANY"] = "DE";
    CountryCodes["GHANA"] = "GH";
    CountryCodes["GIBRALTAR"] = "GI";
    CountryCodes["GREECE"] = "GR";
    CountryCodes["GREENLAND"] = "GL";
    CountryCodes["GRENADA"] = "GD";
    CountryCodes["GUADELOUPE"] = "GP";
    CountryCodes["GUAM"] = "GU";
    CountryCodes["GUATEMALA"] = "GT";
    CountryCodes["GUINEA"] = "GN";
    CountryCodes["GUINEA-BISSAU"] = "GW";
    CountryCodes["GUYANA"] = "GY";
    CountryCodes["HAITI"] = "HT";
    CountryCodes["HEARD ISLAND AND MCDONALD ISLANDS"] = "HM";
    CountryCodes["HONDURAS"] = "HN";
    CountryCodes["HONG KONG"] = "HK";
    CountryCodes["HUNGARY"] = "HU";
    CountryCodes["ICELAND"] = "IS";
    CountryCodes["INDIA"] = "IN";
    CountryCodes["INDONESIA"] = "ID";
    CountryCodes["IRAN, ISLAMIC REPUBLIC OF"] = "IR";
    CountryCodes["IRAQ"] = "IQ";
    CountryCodes["IRELAND"] = "IE";
    CountryCodes["ISRAEL"] = "IL";
    CountryCodes["ITALY"] = "IT";
    CountryCodes["JAMAICA"] = "JM";
    CountryCodes["JAPAN"] = "JP";
    CountryCodes["JORDAN"] = "JO";
    CountryCodes["KAZAKHSTAN"] = "KZ";
    CountryCodes["KENYA"] = "KE";
    CountryCodes["KIRIBATI"] = "KI";
    CountryCodes["KOREA, DEMOCRATIC PEOPLE'S REPUBLIC OF"] = "KP";
    CountryCodes["KOREA, REPUBLIC OF"] = "KR";
    CountryCodes["KUWAIT"] = "KW";
    CountryCodes["KYRGYZSTAN"] = "KG";
    CountryCodes["LAO PEOPLE'S DEMOCRATIC REPUBLIC (LAOS)"] = "LA";
    CountryCodes["LATVIA"] = "LV";
    CountryCodes["LEBANON"] = "LB";
    CountryCodes["LESOTHO"] = "LS";
    CountryCodes["LIBERIA"] = "LR";
    CountryCodes["LIBYA, STATE OF"] = "LY";
    CountryCodes["LIECHTENSTEIN"] = "LI";
    CountryCodes["LITHUANIA"] = "LT";
    CountryCodes["LUXEMBOURG"] = "LU";
    CountryCodes["MACAO"] = "MO";
    CountryCodes["MACEDONIA, THE FORMER YUGOSLAV REPUBLIC OF"] = "MK";
    CountryCodes["MADAGASCAR"] = "MG";
    CountryCodes["MALAWI"] = "MW";
    CountryCodes["MALAYSIA"] = "MY";
    CountryCodes["MALDIVES"] = "MV";
    CountryCodes["MALI"] = "ML";
    CountryCodes["MALTA"] = "MT";
    CountryCodes["MARSHALL ISLANDS"] = "MH";
    CountryCodes["MARTINIQUE"] = "MQ";
    CountryCodes["MAURITANIA"] = "MR";
    CountryCodes["MAURITIUS"] = "MU";
    CountryCodes["MAYOTTE"] = "YT";
    CountryCodes["MEXICO"] = "MX";
    CountryCodes["MICRONESIA, FEDERATED STATES OF"] = "FM";
    CountryCodes["MOLDOVA, REPUBLIC OF"] = "MD";
    CountryCodes["MONACO"] = "MC";
    CountryCodes["MONGOLIA"] = "MN";
    CountryCodes["MONTENEGRO"] = "ME";
    CountryCodes["MONTSERRAT"] = "MS";
    CountryCodes["MOROCCO"] = "MA";
    CountryCodes["MOZAMBIQUE"] = "MZ";
    CountryCodes["MYANMAR"] = "MM";
    CountryCodes["NAMIBIA"] = "NA";
    CountryCodes["NAURU"] = "NR";
    CountryCodes["NEPAL, FEDERAL DEMOCRATIC REPUBLIC OF"] = "NP";
    CountryCodes["NETHERLANDS"] = "NL";
    CountryCodes["NETHERLANDS ANTILLES"] = "AN";
    CountryCodes["NEW CALEDONIA"] = "NC";
    CountryCodes["NEW ZEALAND"] = "NZ";
    CountryCodes["NICARAGUA"] = "NI";
    CountryCodes["NIGER"] = "NE";
    CountryCodes["NIGERIA"] = "NG";
    CountryCodes["NIUE"] = "NU";
    CountryCodes["NORFOLK ISLAND"] = "NF";
    CountryCodes["NORTHERN MARIANA ISLANDS"] = "MP";
    CountryCodes["NORWAY"] = "NO";
    CountryCodes["OMAN"] = "OM";
    CountryCodes["PAKISTAN"] = "PK";
    CountryCodes["PALAU"] = "PW";
    CountryCodes["PALESTINE, STATE OF"] = "PS";
    CountryCodes["PANAMA"] = "PA";
    CountryCodes["PAPUA NEW GUINEA"] = "PG";
    CountryCodes["PARAGUAY"] = "PY";
    CountryCodes["PERU"] = "PE";
    CountryCodes["PHILIPPINES"] = "PH";
    CountryCodes["PITCAIRN"] = "PN";
    CountryCodes["POLAND"] = "PL";
    CountryCodes["PORTUGAL"] = "PT";
    CountryCodes["PUERTO RICO"] = "PR";
    CountryCodes["QATAR"] = "QA";
    CountryCodes["R\u00C9UNION"] = "RE";
    CountryCodes["ROMANIA"] = "RO";
    CountryCodes["RUSSIAN FEDERATION"] = "RU";
    CountryCodes["RWANDA"] = "RW";
    CountryCodes["SAINT HELENA"] = "SH";
    CountryCodes["SAINT KITTS AND NEVIS"] = "KN";
    CountryCodes["SAINT LUCIA"] = "LC";
    CountryCodes["SAINT PIERRE AND MIQUELON"] = "PM";
    CountryCodes["SAINT VINCENT AND THE GRENADINES"] = "VC";
    CountryCodes["SAMOA"] = "WS";
    CountryCodes["SAN MARINO"] = "SM";
    CountryCodes["SAO TOME AND PRINCIPE"] = "ST";
    CountryCodes["SAUDI ARABIA"] = "SA";
    CountryCodes["SENEGAL"] = "SN";
    CountryCodes["SERBIA"] = "RS";
    CountryCodes["SEYCHELLES"] = "SC";
    CountryCodes["SIERRA LEONE"] = "SL";
    CountryCodes["SINGAPORE"] = "SG";
    CountryCodes["SLOVAKIA"] = "SK";
    CountryCodes["SLOVENIA"] = "SI";
    CountryCodes["SOLOMON ISLANDS"] = "SB";
    CountryCodes["SOMALIA"] = "SO";
    CountryCodes["SOUTH AFRICA"] = "ZA";
    CountryCodes["SOUTH GEORGIA AND THE SOUTH SANDWICH ISLANDS"] = "GS";
    CountryCodes["SOUTH SUDAN"] = "SS";
    CountryCodes["SPAIN"] = "ES";
    CountryCodes["SRI LANKA"] = "LK";
    CountryCodes["SUDAN"] = "SD";
    CountryCodes["SURINAME"] = "SR";
    CountryCodes["SVALBARD AND JAN MAYEN"] = "SJ";
    CountryCodes["SWAZILAND"] = "SZ";
    CountryCodes["SWEDEN"] = "SE";
    CountryCodes["SWITZERLAND"] = "CH";
    CountryCodes["SYRIAN ARAB REPUBLIC"] = "SY";
    CountryCodes["TAIWAN"] = "TW";
    CountryCodes["TAJIKISTAN"] = "TJ";
    CountryCodes["TANZANIA, UNITED REPUBLIC OF"] = "TZ";
    CountryCodes["THAILAND"] = "TH";
    CountryCodes["TIMOR-LESTE"] = "TL";
    CountryCodes["TOGO"] = "TG";
    CountryCodes["TOKELAU"] = "TK";
    CountryCodes["TONGA"] = "TO";
    CountryCodes["TRINIDAD AND TOBAGO"] = "TT";
    CountryCodes["TUNISIA"] = "TN";
    CountryCodes["TURKEY"] = "TR";
    CountryCodes["TURKMENISTAN"] = "TM";
    CountryCodes["TURKS AND CAICOS ISLANDS"] = "TC";
    CountryCodes["TUVALU"] = "TV";
    CountryCodes["UGANDA"] = "UG";
    CountryCodes["UKRAINE"] = "UA";
    CountryCodes["UNITED ARAB EMIRATES"] = "AE";
    CountryCodes["UNITED KINGDOM"] = "GB";
    CountryCodes["UNITED STATES"] = "US";
    CountryCodes["UNITED STATES MINOR OUTLYING ISLANDS"] = "UM";
    CountryCodes["URUGUAY"] = "UY";
    CountryCodes["UZBEKISTAN"] = "UZ";
    CountryCodes["VANUATU"] = "VU";
    CountryCodes["VENEZUELA"] = "VE";
    CountryCodes["VIET NAM"] = "VN";
    CountryCodes["VIRGIN ISLANDS, BRITISH"] = "VG";
    CountryCodes["VIRGIN ISLANDS, U.S."] = "VI";
    CountryCodes["WALLIS AND FUTUNA"] = "WF";
    CountryCodes["WESTERN SAHARA"] = "EH";
    CountryCodes["YEMEN"] = "YE";
    CountryCodes["ZAMBIA"] = "ZM";
    CountryCodes["ZIMBABWE"] = "ZW";
})(CountryCodes = exports.CountryCodes || (exports.CountryCodes = {}));
