// @marcapo/modulesession/lib/src/interfaces/languagecodes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LanguageCodes = void 0;
var LanguageCodes;
(function (LanguageCodes) {
    LanguageCodes["Abkhazian"] = "ab";
    LanguageCodes["Afar"] = "aa";
    LanguageCodes["Afrikaans"] = "af";
    LanguageCodes["Akan"] = "ak";
    LanguageCodes["Albanian"] = "sq";
    LanguageCodes["Amharic"] = "am";
    LanguageCodes["Arabic"] = "ar";
    LanguageCodes["Aragonese"] = "an";
    LanguageCodes["Armenian"] = "hy";
    LanguageCodes["Assamese"] = "as";
    LanguageCodes["Avaric"] = "av";
    LanguageCodes["Avestan"] = "ae";
    LanguageCodes["Aymara"] = "ay";
    LanguageCodes["Azerbaijani"] = "az";
    LanguageCodes["Bambara"] = "bm";
    LanguageCodes["Bashkir"] = "ba";
    LanguageCodes["Basque"] = "eu";
    LanguageCodes["Belarusian"] = "be";
    LanguageCodes["Bengali (Bangla)"] = "bn";
    LanguageCodes["Bihari"] = "bh";
    LanguageCodes["Bislama"] = "bi";
    LanguageCodes["Bosnian"] = "bs";
    LanguageCodes["Breton"] = "br";
    LanguageCodes["Bulgarian"] = "bg";
    LanguageCodes["Burmese"] = "my";
    LanguageCodes["Catalan"] = "ca";
    LanguageCodes["Chamorro"] = "ch";
    LanguageCodes["Chechen"] = "ce";
    LanguageCodes["Chichewa, Chewa, Nyanja"] = "ny";
    LanguageCodes["Chinese"] = "zh";
    LanguageCodes["Chinese (Simplified)"] = "zh-Hans";
    LanguageCodes["Chinese (Traditional)"] = "zh-Hant";
    LanguageCodes["Chuvash"] = "cv";
    LanguageCodes["Cornish"] = "kw";
    LanguageCodes["Corsican"] = "co";
    LanguageCodes["Cree"] = "cr";
    LanguageCodes["Croatian"] = "hr";
    LanguageCodes["Czech"] = "cs";
    LanguageCodes["Danish"] = "da";
    LanguageCodes["Divehi, Dhivehi, Maldivian"] = "dv";
    LanguageCodes["Dutch"] = "nl";
    LanguageCodes["Dzongkha"] = "dz";
    LanguageCodes["English"] = "en";
    LanguageCodes["Esperanto"] = "eo";
    LanguageCodes["Estonian"] = "et";
    LanguageCodes["Ewe"] = "ee";
    LanguageCodes["Faroese"] = "fo";
    LanguageCodes["Fijian"] = "fj";
    LanguageCodes["Finnish"] = "fi";
    LanguageCodes["French"] = "fr";
    LanguageCodes["Fula, Fulah, Pulaar, Pular"] = "ff";
    LanguageCodes["Galician"] = "gl";
    LanguageCodes["Gaelic (Scottish)"] = "gd";
    LanguageCodes["Gaelic (Manx)"] = "gv";
    LanguageCodes["Georgian"] = "ka";
    LanguageCodes["German"] = "de";
    LanguageCodes["Greek"] = "el";
    LanguageCodes["Greenlandic"] = "kl";
    LanguageCodes["Guarani"] = "gn";
    LanguageCodes["Gujarati"] = "gu";
    LanguageCodes["Haitian Creole"] = "ht";
    LanguageCodes["Hausa"] = "ha";
    LanguageCodes["Hebrew"] = "he";
    LanguageCodes["Herero"] = "hz";
    LanguageCodes["Hindi"] = "hi";
    LanguageCodes["Hiri Motu"] = "ho";
    LanguageCodes["Hungarian"] = "hu";
    LanguageCodes["Icelandic"] = "is";
    LanguageCodes["Ido"] = "io";
    LanguageCodes["Igbo"] = "ig";
    LanguageCodes["Indonesianid, "] = "in";
    LanguageCodes["Interlingua"] = "ia";
    LanguageCodes["Interlingue"] = "ie";
    LanguageCodes["Inuktitut"] = "iu";
    LanguageCodes["Inupiak"] = "ik";
    LanguageCodes["Irish"] = "ga";
    LanguageCodes["Italian"] = "it";
    LanguageCodes["Japanese"] = "ja";
    LanguageCodes["Javanese"] = "jv";
    LanguageCodes["Kalaallisut, Greenlandic"] = "kl";
    LanguageCodes["Kannada"] = "kn";
    LanguageCodes["Kanuri"] = "kr";
    LanguageCodes["Kashmiri"] = "ks";
    LanguageCodes["Kazakh"] = "kk";
    LanguageCodes["Khmer"] = "km";
    LanguageCodes["Kikuyu"] = "ki";
    LanguageCodes["Kinyarwanda (Rwanda)"] = "rw";
    LanguageCodes["Kirundi"] = "rn";
    LanguageCodes["Kyrgyz"] = "ky";
    LanguageCodes["Komi"] = "kv";
    LanguageCodes["Kongo"] = "kg";
    LanguageCodes["Korean"] = "ko";
    LanguageCodes["Kurdish"] = "ku";
    LanguageCodes["Kwanyama"] = "kj";
    LanguageCodes["Lao"] = "lo";
    LanguageCodes["Latin"] = "la";
    LanguageCodes["Latvian (Lettish)"] = "lv";
    LanguageCodes["Limburgish ( Limburger)"] = "li";
    LanguageCodes["Lingala"] = "ln";
    LanguageCodes["Lithuanian"] = "lt";
    LanguageCodes["Luga-Katanga"] = "lu";
    LanguageCodes["Luganda, Ganda"] = "lg";
    LanguageCodes["Luxembourgish"] = "lb";
    LanguageCodes["Manx"] = "gv";
    LanguageCodes["Macedonian"] = "mk";
    LanguageCodes["Malagasy"] = "mg";
    LanguageCodes["Malay"] = "ms";
    LanguageCodes["Malayalam"] = "ml";
    LanguageCodes["Maltese"] = "mt";
    LanguageCodes["Maori"] = "mi";
    LanguageCodes["Marathi"] = "mr";
    LanguageCodes["Marshallese"] = "mh";
    LanguageCodes["Moldavian"] = "mo";
    LanguageCodes["Mongolian"] = "mn";
    LanguageCodes["Nauru"] = "na";
    LanguageCodes["Navajo"] = "nv";
    LanguageCodes["Ndonga"] = "ng";
    LanguageCodes["Northern Ndebele"] = "nd";
    LanguageCodes["Nepali"] = "ne";
    LanguageCodes["Norwegian"] = "no";
    LanguageCodes["Norwegian bokm\u00E5l"] = "nb";
    LanguageCodes["Norwegian nynorsk"] = "nn";
    LanguageCodes["Nuosu"] = "ii";
    LanguageCodes["Occitan"] = "oc";
    LanguageCodes["Ojibwe"] = "oj";
    LanguageCodes["Old Church Slavonic, Old Bulgarian"] = "cu";
    LanguageCodes["Oriya"] = "or";
    LanguageCodes["Oromo (Afaan Oromo)"] = "om";
    LanguageCodes["Ossetian"] = "os";
    LanguageCodes["P\u0101li"] = "pi";
    LanguageCodes["Pashto, Pushto"] = "ps";
    LanguageCodes["Persian (Farsi)"] = "fa";
    LanguageCodes["Polish"] = "pl";
    LanguageCodes["Portuguese"] = "pt";
    LanguageCodes["Punjabi (Eastern)"] = "pa";
    LanguageCodes["Quechua"] = "qu";
    LanguageCodes["Romansh"] = "rm";
    LanguageCodes["Romanian"] = "ro";
    LanguageCodes["Russian"] = "ru";
    LanguageCodes["Sami"] = "se";
    LanguageCodes["Samoan"] = "sm";
    LanguageCodes["Sango"] = "sg";
    LanguageCodes["Sanskrit"] = "sa";
    LanguageCodes["Serbian"] = "sr";
    LanguageCodes["Serbo-Croatian"] = "sh";
    LanguageCodes["Sesotho"] = "st";
    LanguageCodes["Setswana"] = "tn";
    LanguageCodes["Shona"] = "sn";
    LanguageCodes["Sichuan Yi"] = "ii";
    LanguageCodes["Sindhi"] = "sd";
    LanguageCodes["Sinhalese"] = "si";
    LanguageCodes["Siswati"] = "ss";
    LanguageCodes["Slovak"] = "sk";
    LanguageCodes["Slovenian"] = "sl";
    LanguageCodes["Somali"] = "so";
    LanguageCodes["Southern Ndebele"] = "nr";
    LanguageCodes["Spanish"] = "es";
    LanguageCodes["Sundanese"] = "su";
    LanguageCodes["Swahili (Kiswahili)"] = "sw";
    LanguageCodes["Swati"] = "ss";
    LanguageCodes["Swedish"] = "sv";
    LanguageCodes["Tagalog"] = "tl";
    LanguageCodes["Tahitian"] = "ty";
    LanguageCodes["Tajik"] = "tg";
    LanguageCodes["Tamil"] = "ta";
    LanguageCodes["Tatar"] = "tt";
    LanguageCodes["Telugu"] = "te";
    LanguageCodes["Thai"] = "th";
    LanguageCodes["Tibetan"] = "bo";
    LanguageCodes["Tigrinya"] = "ti";
    LanguageCodes["Tonga"] = "to";
    LanguageCodes["Tsonga"] = "ts";
    LanguageCodes["Turkish"] = "tr";
    LanguageCodes["Turkmen"] = "tk";
    LanguageCodes["Twi"] = "tw";
    LanguageCodes["Uyghur"] = "ug";
    LanguageCodes["Ukrainian"] = "uk";
    LanguageCodes["Urdu"] = "ur";
    LanguageCodes["Uzbek"] = "uz";
    LanguageCodes["Venda"] = "ve";
    LanguageCodes["Vietnamese"] = "vi";
    LanguageCodes["Volap\u00FCk"] = "vo";
    LanguageCodes["Wallon"] = "wa";
    LanguageCodes["Welsh"] = "cy";
    LanguageCodes["Wolof"] = "wo";
    LanguageCodes["Western Frisian"] = "fy";
    LanguageCodes["Xhosa"] = "xh";
    LanguageCodes["Yiddish"] = "ji";
    LanguageCodes["Yoruba"] = "yo";
    LanguageCodes["Zhuang, Chuang"] = "za";
    LanguageCodes["Zulu"] = "zu";
})(LanguageCodes = exports.LanguageCodes || (exports.LanguageCodes = {}));
